### DISCUSSION TITLE
在IHostService中使用了异步，form界面会诡异的直接崩溃

### DISCUSSION THREAD

**Discussion by oufu99** [BYJRK/DotNet-Discussions#256] [C# 技术交流]
![81978699dc40cffb1f90d0a8286486b](https://github.com/BYJRK/DotNet-Discussions/assets/18545756/bde3634d-e23a-4474-b646-57e7e10caa6f)
为啥加了一个这个  await Task.Delay(2000); 后面运行form的时候就会有问题，界面只要一报错整个系统就崩溃了
把这句注释掉以后，又没问题了，没理解这是为什么。

之前以为是线程没有回到主线程的原因,然后用了大佬课程里面讲的那个GetAwait回到原始线程，但是还是不行

**Comment by CodingOctocat** [comment]
有添加断点调试吗？哪一行开始崩溃？另外，程序报错了不是本来就会崩溃，喵？？

**Comment by SlimeNull** [comment]
你应该处理掉引发的异常，例如某些按钮的事件回调中，都应该做好异常处理，只要程序不会出现 UnhandledException，就不会崩溃了喵

**Comment by freeman888** [comment]
感觉是StartAsync方法就没有在主线程上面运行，await之后自然回不到主线程。
